### PR TITLE
chore(utils, product): Attempt to simplify module scripts export

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
In an attempt to simplify new module scaffolding, some utils have been created to generate the different scripts a module needs
- run migration script can now be exported directly from the util in the index
- revert migration script can now be exported directly from the util in the index
- seed can be directly exported from the scripts/bin directory

### CODE REVIEW

**Comment by adrien2p** [comment]
This is a WIP side task, do not review yet

**Comment by adrien2p** [comment]
@olivermrbl here is my attempt to centralize the modules scripts part. Let me know what is your thinking around it

File: packages/product/src/scripts/bin/run-seed.ts [hunk 1]
```diff
@@ -15,5 +22,16 @@ export default (async () => {
     )
   }
 
+  const run = ModulesSdkUtils.buildSeedScript({
+    moduleName: Modules.PRODUCT,
+    models: ProductModels,
+    pathToMigrations: __dirname + "/../../migrations",
+    executorFn: async (manager, data) => {
```
Comment (olivermrbl) [inline packages/product/src/scripts/bin/run-seed.ts]:
```suggestion
    seedFuncs: async (manager, data) => {
```

File: packages/product/package.json [hunk 2]
```diff
@@ -8,8 +8,6 @@
     "dist"
   ],
   "bin": {
-    "medusa-product-migrations-down": "dist/scripts/bin/run-migration-down.js",
-    "medusa-product-migrations-up": "dist/scripts/bin/run-migration-up.js",
```
Comment (olivermrbl) [inline packages/product/package.json:12]: question: We won't have scripts any longer, only functions?

**Review by olivermrbl — APPROVED** [review]
Think this is a great addition!

**Comment by olivermrbl** [comment]
@pKorsholm @fPolic @riqwan, what do you think?

File: packages/product/package.json (hunk 2 above)
Comment (adrien2p) [reply packages/product/package.json:12]: yes, we use to have script cause it was not yet handled by the CLI, but now the CLI handle them plus I ve fixed the client url loading. So now when you run `medusa migrations run` it goes through those functions as they are grabbed from the module exports

Comment (adrien2p) [reply packages/product/package.json:12]: I ve only kept the seed script as this one is manual and you can pass a file path to it

Comment (adrien2p) [reply packages/product/package.json:12]: I ve run manually in debug to ensure that and it worked pretty well

File: packages/product/src/scripts/bin/run-seed.ts (hunk 1 above)
Comment (adrien2p) [reply packages/product/src/scripts/bin/run-seed.ts]: seedHandler? it is a single handler wdyt?

**Comment by adrien2p** [comment]
If we agree on that, I can migrate the other modules too as part of that pr, wdyt guys?

File: packages/utils/src/modules-sdk/migration-scripts/seed.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    const orm = await mikroOrmCreateConnection(
+      dbData,
+      entities,
+      pathToMigrations
+    )
+    const manager = orm.em.fork()
+
+    try {
+      logger?.info(`Inserting ${moduleName} data...`)
+      executorFn(manager, dataSeed)
```
Comment (riqwan) [inline packages/utils/src/modules-sdk/migration-scripts/seed.ts]: will be helpful for seed functions to have logger as well.

**Review by riqwan — APPROVED** [review]
nnoice! 🔥

File: packages/utils/src/modules-sdk/migration-scripts/seed.ts (hunk 3 above)
Comment (adrien2p) [reply packages/utils/src/modules-sdk/migration-scripts/seed.ts]: true, I ll pass it down 👍

**Review by pKorsholm — APPROVED** [review]
LGTM!

File: packages/utils/src/modules-sdk/migration-scripts/seed.ts (hunk 3 above)
Comment (adrien2p) [reply packages/utils/src/modules-sdk/migration-scripts/seed.ts]: done

**Review by fPolic — APPROVED** [review]
LGTM 💪

File: packages/product/src/scripts/bin/run-seed.ts (hunk 1 above)
Comment (olivermrbl) [reply packages/product/src/scripts/bin/run-seed.ts]: Sure!

**Comment by adrien2p** [comment]
@olivermrbl should we proceed with this one? In that case I ll add the automerge label

**Pull request merged**